compiler/typecheck/TcRnDriver.hs

```diff
@@ -2768,7 +2768,7 @@ ppr_types debug type_env
              -- etc are suppressed (unless -dppr-debug),
              -- because they appear elsehwere
 
-    ppr_sig id = hang (ppr id <+> dcolon) 2 (ppr (tidyTopType (idType id)))
+    ppr_sig id = hang (ppr id <+> of_type) 2 (ppr (tidyTopType (idType id)))
```

Comment:
I think we didn't do the `dcolon` for `of_type` substitution before now for no better reason than we couldn't apply the patch in Nix. It's good that we are getting this fixed.